src/mongo/delegates/Updates.class.php

```diff
@@ -659,6 +668,75 @@ protected function applyChangeSet(\Tripod\ChangeSet $cs, $originalCBDs, $context
         }
     }
 
+    private function getAdditionsRemovalsGroupedByNsPredicate($cs,$additions,$removals)
```

Comment:
More 70% feedback here, but some type hinting on these.  In the interest of this code review, I'm assuming it's ChangeSet, Array, Array